src/lib/comms/sol-http-server-impl-microhttpd.c

```diff
@@ -117,21 +119,28 @@ build_mhd_response(const struct sol_http_response *response)
 
     SOL_HTTP_PARAM_FOREACH_IDX (&response->param, value, idx) {
         int ret;
-        char buffer[512];
+        char buffer[512], key[512], value_str[512];
```

Comment:
I'm worried about this 512 constant. Does these really have such a low maximum defined?

If not, I'd suggest change both buffer/key/value to a `sol_buffer` that is initialized before the loop. Then, inside the loop you keep reseting `buffer.used = 0`. You can then append whatever you want, it will grow the buffer as required.

For key-value, you use the same buffer, just append a null byte:
```c
buf.used = 0; // reset to buffer start, but doesn't free internal memory (keeps capacity)
ret = sol_buffer_append_slice(&buf, value->value.key_value.key);
if (ret < 0) goto err;
ret = sol_buffer_append_char(&buf, 0); // force a real \0, this is accounted as used, so is not overriden by the next append slice
if (ret < 0) goto err;
ret = sol_buffer_append_slice(&buf, value->value.key_value.value); // if buffer is default flags, then we have a '\0' here, no need to force it.
if (ret < 0) goto err;
if (MHD_add_response_header(r, sol_buffer_at(&buf, 0), sol_buffer_at(&buf, value->value.key_value.key.len + 1)) == MHD_NO) {
...
}
```